engine/runtime/src/main/java/org/enso/interpreter/node/callable/resolver/HostMethodCallNode.java

```diff
@@ -150,6 +155,14 @@ public static PolyglotCallType getPolyglotCallType(
       UnresolvedSymbol symbol,
       InteropLibrary library,
       MethodResolverNode methodResolverNode) {
+    if (symbol.getScope().getMethodForPolyglot(self, symbol.getName(), false)
```

Comment:
`getMethodForPolyglot` already returns `Function` type. Thus, the instanceof check here is redundant, and null check should be sufficient.
```suggestion
    if (symbol.getScope().getMethodForPolyglot(self, symbol.getName(), false) != null) { 
```